### PR TITLE
Vulnerability Fixes, Dockerfile Update, and Dependency Addition

### PR DESCRIPTION
**Pull request by SantiagoRojasBuitrago**
This Pull Request addresses the following improvements and fixes in the project:

- Vulnerability fix for 083 (XML injection (XXE)): The vulnerability has been resolved by adding the forbid_external=True configuration to the etree.parse method to prevent access to external resources in XML entities or DTD.

- Dockerfile Update: The line EXPOSE 8080 has been removed from the Dockerfile.nginx-alpine to enhance security and minimize points of exposure.

- Dependency Addition: New dependencies have been added to the requirements.txt file to ensure proper project functionality and keep the libraries up to date.

### CODE REVIEW

File: Dockerfile.nginx-alpine [hunk 1]
```diff
@@ -164,5 +164,5 @@ ENV \
   METRICS_HTTP_AUTH_USER="" \
   METRICS_HTTP_AUTH_PASSWORD=""
 USER ${uid}
-EXPOSE 8080
+
```
Comment (Gaviria9601) [inline Dockerfile.nginx-alpine:167]: Please also add in Dockerfile.nginx-debian

File: requirements.txt [hunk 2]
```diff
@@ -83,3 +83,69 @@ netaddr==0.8.0
 vulners==2.0.10
 fontawesomefree==6.4.0
 azure-devops==7.1.0b3
+
+
+#New Dependencies
```
Comment (Gaviria9601) [inline requirements.txt:88]: add dependency azure-core==1.26.4

File: requirements.txt [hunk 3]
```diff
@@ -83,3 +83,69 @@ netaddr==0.8.0
...
+MarkupSafe==2.0
```
Comment (Gaviria9601) [inline requirements.txt]: Why 2.0 . Should be 2.1.2

File: requirements.txt [hunk 4]
```diff
@@ -83,3 +83,69 @@ netaddr==0.8.0
...
+prompt-toolkit==3.0.20
```
Comment (Gaviria9601) [inline requirements.txt]: shoud be 3.0.38

File: requirements.txt [hunk 5]
```diff
@@ -83,3 +83,69 @@ netaddr==0.8.0
...
+billiard==3.6.4.0
+inflection==0.5.1
+requests-toolbelt==0.10.1
+smmap==4.0.0
```
Comment (Gaviria9601) [inline requirements.txt]: should be 5.0.0

File: requirements.txt (hunk 2 above)
Comment (Gaviria9601) [inline requirements.txt:88]: Please review version according to dependencies tree.

**Review by Gaviria9601 — COMMENTED** [review]
Please, review the comments.

File: Dockerfile.nginx-alpine (hunk 1 above)
Comment (SantiagoRojasBuitrago) [reply Dockerfile.nginx-alpine:167]: Solved

File: requirements.txt (hunk 2 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt:88]: Added

File: requirements.txt (hunk 5 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: docker-compose.override.unit_tests_cicd.yml [hunk 6]
```diff
@@ -62,7 +62,28 @@ services:
   redis:
     image: busybox:1.36.1-musl
     entrypoint: ['echo', 'skipping', 'redis']
+
 volumes:
-  defectdojo_data_unit_tests: {}
-  defectdojo_postgres_unit_tests: {}
-  defectdojo_media_unit_tests: {}
+  defectdojo_data_unit_tests:
```
Comment (Gaviria9601) [inline docker-compose.override.unit_tests_cicd.yml]: Remove these changes. The docker-compose is out of scope.

File: requirements.txt [hunk 7]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+ruamel.yaml.clib==0.2.6
+asgiref==3.5.2
+ruamel.yaml==0.17.16
+itypes==1.2.0
+elementpath==2.3.2
+requests-oauthlib==1.3.0
+vine==5.0.0
+wrapt==1.13.2
+appdirs==1.4.4
+amqp==5.0.9
```
Comment (Gaviria9601) [inline requirements.txt]: should be 5.1.1

File: requirements.txt [hunk 8]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+PyYAML==6.0
+click-plugins==1.1.1
+ruamel.yaml.clib==0.2.6
+asgiref==3.5.2
+ruamel.yaml==0.17.16
+itypes==1.2.0
+elementpath==2.3.2
+requests-oauthlib==1.3.0
+vine==5.0.0
+wrapt==1.13.2
```
Comment (Gaviria9601) [inline requirements.txt]: should be 1.15.0

File: requirements.txt [hunk 9]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+gitdb==4.0.7
+Jinja2==3.0.2
+PyYAML==6.0
+click-plugins==1.1.1
+ruamel.yaml.clib==0.2.6
+asgiref==3.5.2
+ruamel.yaml==0.17.16
+itypes==1.2.0
+elementpath==2.3.2
+requests-oauthlib==1.3.0
```
Comment (Gaviria9601) [inline requirements.txt]: should be 1.3.1

File: requirements.txt [hunk 10]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+cffi==1.15.0
+gitdb==4.0.7
+Jinja2==3.0.2
+PyYAML==6.0
+click-plugins==1.1.1
+ruamel.yaml.clib==0.2.6
+asgiref==3.5.2
+ruamel.yaml==0.17.16
+itypes==1.2.0
+elementpath==2.3.2
```
Comment (Gaviria9601) [inline requirements.txt]: should be 4.1.0

File: requirements.txt [hunk 11]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+xmlschema==1.8.0
+sqlparse==0.4.2
+cffi==1.15.0
+gitdb==4.0.7
+Jinja2==3.0.2
+PyYAML==6.0
+click-plugins==1.1.1
+ruamel.yaml.clib==0.2.6
+asgiref==3.5.2
+ruamel.yaml==0.17.16
```
Comment (Gaviria9601) [inline requirements.txt]: should be 0.17.21

File: requirements.txt [hunk 12]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+charset-normalizer==2.0.7
+xmlschema==1.8.0
+sqlparse==0.4.2
+cffi==1.15.0
+gitdb==4.0.7
+Jinja2==3.0.2
+PyYAML==6.0
+click-plugins==1.1.1
+ruamel.yaml.clib==0.2.6
+asgiref==3.5.2
```
Comment (Gaviria9601) [inline requirements.txt]: should be 3.6.0

File: requirements.txt [hunk 13]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+rsa==4.7.2
+python3-openid==3.2.0
+certifi==2021.10.8
+PyNaCl==1.4.0
+charset-normalizer==2.0.7
+xmlschema==1.8.0
+sqlparse==0.4.2
+cffi==1.15.0
+gitdb==4.0.7
+Jinja2==3.0.2
```
Comment (Gaviria9601) [inline requirements.txt]: should be 3.1.2

File: requirements.txt [hunk 14]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+jsonschema==4.1.1
+rsa==4.7.2
+python3-openid==3.2.0
+certifi==2021.10.8
+PyNaCl==1.4.0
+charset-normalizer==2.0.7
+xmlschema==1.8.0
+sqlparse==0.4.2
+cffi==1.15.0
+gitdb==4.0.7
```
Comment (Gaviria9601) [inline requirements.txt]: should 4.0.10

File: requirements.txt [hunk 15]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+click==8.0.3
+jsonschema==4.1.1
+rsa==4.7.2
+python3-openid==3.2.0
+certifi==2021.10.8
+PyNaCl==1.4.0
+charset-normalizer==2.0.7
+xmlschema==1.8.0
+sqlparse==0.4.2
+cffi==1.15.0
```
Comment (Gaviria9601) [inline requirements.txt]: should 1.15.1

File: requirements.txt [hunk 16]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+django-appconf==1.0.5
+click==8.0.3
+jsonschema==4.1.1
+rsa==4.7.2
+python3-openid==3.2.0
+certifi==2021.10.8
+PyNaCl==1.4.0
+charset-normalizer==2.0.7
+xmlschema==1.8.0
+sqlparse==0.4.2
```
Comment (Gaviria9601) [inline requirements.txt]: should be 0.4.3

File: requirements.txt [hunk 17]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+yarl==1.7.0
+django-appconf==1.0.5
+click==8.0.3
+jsonschema==4.1.1
+rsa==4.7.2
+python3-openid==3.2.0
+certifi==2021.10.8
+PyNaCl==1.4.0
+charset-normalizer==2.0.7
+xmlschema==1.8.0
```
Comment (Gaviria9601) [inline requirements.txt]: should be 2.2.2

File: requirements.txt [hunk 18]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+pilkit==2.0
+yarl==1.7.0
+django-appconf==1.0.5
+click==8.0.3
+jsonschema==4.1.1
+rsa==4.7.2
+python3-openid==3.2.0
+certifi==2021.10.8
+PyNaCl==1.4.0
+charset-normalizer==2.0.7
```
Comment (Gaviria9601) [inline requirements.txt]: should be 3.1.0

File: requirements.txt [hunk 19]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+uritemplate==3.0.1
+pilkit==2.0
+yarl==1.7.0
+django-appconf==1.0.5
+click==8.0.3
+jsonschema==4.1.1
+rsa==4.7.2
+python3-openid==3.2.0
+certifi==2021.10.8
+PyNaCl==1.4.0
```
Comment (Gaviria9601) [inline requirements.txt]: should be 1.5.0

File: requirements.txt [hunk 20]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+webencodings==0.5.1
+uritemplate==3.0.1
+pilkit==2.0
+yarl==1.7.0
+django-appconf==1.0.5
+click==8.0.3
+jsonschema==4.1.1
+rsa==4.7.2
+python3-openid==3.2.0
+certifi==2021.10.8
```
Comment (Gaviria9601) [inline requirements.txt]: should be 2022.12.7

File: requirements.txt [hunk 21]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+Deprecated==1.2.13
+click-didyoumean==0.3.0
+webencodings==0.5.1
+uritemplate==3.0.1
+pilkit==2.0
+yarl==1.7.0
+django-appconf==1.0.5
+click==8.0.3
+jsonschema==4.1.1
+rsa==4.7.2
```
Comment (Gaviria9601) [inline requirements.txt]: should be 4.9

File: requirements.txt [hunk 22]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+prometheus-client==0.11.0
+Deprecated==1.2.13
+click-didyoumean==0.3.0
+webencodings==0.5.1
+uritemplate==3.0.1
+pilkit==2.0
+yarl==1.7.0
+django-appconf==1.0.5
+click==8.0.3
+jsonschema==4.1.1
```
Comment (Gaviria9601) [inline requirements.txt]: should be 4.17.3

File: requirements.txt [hunk 23]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+ecdsa==0.17.0
+prometheus-client==0.11.0
+Deprecated==1.2.13
+click-didyoumean==0.3.0
+webencodings==0.5.1
+uritemplate==3.0.1
+pilkit==2.0
+yarl==1.7.0
+django-appconf==1.0.5
+click==8.0.3
```
Comment (Gaviria9601) [inline requirements.txt]: should be 8.1.3

File: requirements.txt [hunk 24]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+multidict==5.2.0
+pyasn1==0.4.8
+ecdsa==0.17.0
+prometheus-client==0.11.0
+Deprecated==1.2.13
+click-didyoumean==0.3.0
+webencodings==0.5.1
+uritemplate==3.0.1
+pilkit==2.0
+yarl==1.7.0
```
Comment (Gaviria9601) [inline requirements.txt]: should be 1.8.2

File: requirements.txt [hunk 25]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+google-auth-oauthlib==0.4.6
+coreschema==0.0.4
+multidict==5.2.0
+pyasn1==0.4.8
+ecdsa==0.17.0
+prometheus-client==0.11.0
+Deprecated==1.2.13
+click-didyoumean==0.3.0
+webencodings==0.5.1
+uritemplate==3.0.1
```
Comment (Gaviria9601) [inline requirements.txt]: should 4.1.1

File: requirements.txt [hunk 26]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+pyrsistent==0.18.0
+click-repl==0.2.0
+pycparser==2.20
+et-xmlfile==1.1.0
+google-auth-oauthlib==0.4.6
+coreschema==0.0.4
+multidict==5.2.0
+pyasn1==0.4.8
+ecdsa==0.17.0
+prometheus-client==0.11.0
```
Comment (Gaviria9601) [inline requirements.txt]: should be 0.16.0

File: requirements.txt [hunk 27]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+pysaml2==7.0.1
+pyrsistent==0.18.0
+click-repl==0.2.0
+pycparser==2.20
+et-xmlfile==1.1.0
+google-auth-oauthlib==0.4.6
+coreschema==0.0.4
+multidict==5.2.0
+pyasn1==0.4.8
+ecdsa==0.17.0
```
Comment (Gaviria9601) [inline requirements.txt]: should be 0.18.0

File: requirements.txt [hunk 28]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+idna==2.10
+packaging==21.0
+pysaml2==7.0.1
+pyrsistent==0.18.0
+click-repl==0.2.0
+pycparser==2.20
+et-xmlfile==1.1.0
+google-auth-oauthlib==0.4.6
+coreschema==0.0.4
+multidict==5.2.0
```
Comment (Gaviria9601) [inline requirements.txt]: should be 6.0.4

File: requirements.txt [hunk 29]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+six==1.16.0
+attrs==21.2.0
+idna==2.10
+packaging==21.0
+pysaml2==7.0.1
+pyrsistent==0.18.0
+click-repl==0.2.0
+pycparser==2.20
+et-xmlfile==1.1.0
+google-auth-oauthlib==0.4.6
```
Comment (Gaviria9601) [inline requirements.txt]: should be 1.0.0

File: requirements.txt [hunk 30]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+kombu==5.2.3
+python-jose==3.3.0
+six==1.16.0
+attrs==21.2.0
+idna==2.10
+packaging==21.0
+pysaml2==7.0.1
+pyrsistent==0.18.0
+click-repl==0.2.0
+pycparser==2.20
```
Comment (Gaviria9601) [inline requirements.txt]: should be 2.21

File: requirements.txt [hunk 31]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+wcwidth==0.2.5
+#pyOpenSSL==21.0.0
+kombu==5.2.3
+python-jose==3.3.0
+six==1.16.0
+attrs==21.2.0
+idna==2.10
+packaging==21.0
+pysaml2==7.0.1
+pyrsistent==0.18.0
```
Comment (Gaviria9601) [inline requirements.txt]: should be 0.19.3

File: requirements.txt [hunk 32]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+smmap==4.0.0
+wcwidth==0.2.5
+#pyOpenSSL==21.0.0
+kombu==5.2.3
+python-jose==3.3.0
+six==1.16.0
+attrs==21.2.0
+idna==2.10
+packaging==21.0
+pysaml2==7.0.1
```
Comment (Gaviria9601) [inline requirements.txt]: shoould be 7.4.1

File: requirements.txt [hunk 33]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+requests-toolbelt==0.10.1
+smmap==4.0.0
+wcwidth==0.2.5
+#pyOpenSSL==21.0.0
+kombu==5.2.3
+python-jose==3.3.0
+six==1.16.0
+attrs==21.2.0
+idna==2.10
+packaging==21.0
```
Comment (Gaviria9601) [inline requirements.txt]: should be 23.0

File: requirements.txt [hunk 34]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+inflection==0.5.1
+requests-toolbelt==0.10.1
+smmap==4.0.0
+wcwidth==0.2.5
+#pyOpenSSL==21.0.0
+kombu==5.2.3
+python-jose==3.3.0
+six==1.16.0
+attrs==21.2.0
+idna==2.10
```
Comment (Gaviria9601) [inline requirements.txt]: should be 3.4

File: requirements.txt [hunk 35]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+billiard==3.6.4.0
+inflection==0.5.1
+requests-toolbelt==0.10.1
+smmap==4.0.0
+wcwidth==0.2.5
+#pyOpenSSL==21.0.0
+kombu==5.2.3
+python-jose==3.3.0
+six==1.16.0
+attrs==21.2.0
```
Comment (Gaviria9601) [inline requirements.txt]: should be 22.2.0

File: requirements.txt [hunk 36]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+azure-core==1.26.4
+MarkupSafe==2.1.2
+prompt-toolkit==3.0.38
+billiard==3.6.4.0
+inflection==0.5.1
+requests-toolbelt==0.10.1
+smmap==4.0.0
+wcwidth==0.2.5
+#pyOpenSSL==21.0.0
+kombu==5.2.3
```
Comment (Gaviria9601) [inline requirements.txt]: should be 5.2.4

File: requirements.txt [hunk 37]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+#New Dependencies
+azure-core==1.26.4
+MarkupSafe==2.1.2
+prompt-toolkit==3.0.38
+billiard==3.6.4.0
+inflection==0.5.1
+requests-toolbelt==0.10.1
+smmap==4.0.0
+wcwidth==0.2.5
+#pyOpenSSL==21.0.0
```
Comment (Gaviria9601) [inline requirements.txt]: delete the line

File: requirements.txt [hunk 38]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+
+#New Dependencies
+azure-core==1.26.4
+MarkupSafe==2.1.2
+prompt-toolkit==3.0.38
+billiard==3.6.4.0
+inflection==0.5.1
+requests-toolbelt==0.10.1
+smmap==4.0.0
+wcwidth==0.2.5
```
Comment (Gaviria9601) [inline requirements.txt]: should be 0.2.6

File: requirements.txt [hunk 39]
```diff
@@ -83,3 +83,70 @@ netaddr==0.8.0
...
+elementpath==2.3.2
+requests-oauthlib==1.3.0
+vine==5.0.0
+wrapt==1.13.2
+appdirs==1.4.4
+amqp==5.0.9
+coreapi==2.3.3
+msrest==0.7.1
+tinycss2==1.1.1
+typing-extensions==4.6.3
```
Comment (Gaviria9601) [inline requirements.txt:148]: because this dependency ?

**Review by Gaviria9601 — COMMENTED** [review]
New comments

File: requirements.txt (hunk 3 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 4 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: docker-compose.override.unit_tests_cicd.yml (hunk 6 above)
Comment (SantiagoRojasBuitrago) [reply docker-compose.override.unit_tests_cicd.yml]: Changes removed

File: requirements.txt (hunk 7 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 8 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 10 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 11 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 12 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 13 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 14 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 16 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 17 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 18 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 19 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 21 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 23 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 24 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 25 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 26 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 27 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 29 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 30 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 31 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 22 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 20 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 15 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 28 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 32 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 33 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 34 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 35 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 36 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 37 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Removed

File: requirements.txt (hunk 38 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt]: Updated

File: requirements.txt (hunk 39 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt:148]: The dependency is requested in some XML files

File: requirements.txt (hunk 2 above)
Comment (SantiagoRojasBuitrago) [reply requirements.txt:88]: Reviewed

**Review by Gaviria9601 — APPROVED** [review]
Thanks for fixes.

**Pull request merged**